RecoLocalCalo/HcalRecAlgos/src/HLTv2.cc

```diff
@@ -0,0 +1,189 @@
+  Float_t n4=0;
+  getLandauFrac(-tsShift4+25,-tsShift4+50,n4);
+
+  Float_t i5=0;
+  getLandauFrac(-tsShift5,-tsShift5+25,i5);
+  Float_t n5=0;
+  getLandauFrac(-tsShift5+25,-tsShift5+50,n5);
+
+  Float_t ch3=corrCharge[3]/i3;
+  Float_t ch4=(i3*corrCharge[4]-n3*corrCharge[3])/(i3*i4);
+  Float_t ch5=(n3*n4*corrCharge[3]-i4*nn3*corrCharge[3]-i3*n4*corrCharge[4]+i3*i4*corrCharge[5])/(i3*i4*i5);
+
+  if (ch3<-3 && fNegStrat==HLTv2::ReqPos) {
+    ch3=-3;
+    ch4=corrCharge[4]/i4;
+    ch5=(i4*corrCharge[5]-n4*corrCharge[4])/(i4*i5);
+  }
+  
+  if (ch5<-3 && fNegStrat==HLTv2::ReqPos) {
+    ch4=ch4+(ch5+3);
+    ch5=-3;
+  }
+
+  if (fNegStrat==HLTv2::FromGreg) {
+    if (ch3<-3) {
+      ch3=-3;
+      ch4=corrCharge[4]/i4;
+      ch5=(i4*corrCharge[5]-n4*corrCharge[4])/(i4*i5);
+    }
+    if (ch5<-3 && ch4>15) {
+      double ratio = (corrCharge[4]-ch3*i3)/(corrCharge[5]+3*i5);
+      if (ratio < 5 && ratio > 0.5) {
+	double invG = -13.11+11.29*TMath::Sqrt(2*TMath::Log(5.133/ratio));
+	Float_t iG=0;
+	getLandauFrac(-invG,-invG+25,iG);
+	ch4=(corrCharge[4]-ch3*n3)/(iG);
+	ch5=-3;
+	tsShift4=invG;
+      }
+    }
+  }
+  
+  if (ch3<1) {// && (fNegStrat==HLTv2::ReqPos || fNegStrat==HLTv2::FromGreg)) {
+    ch3=0;
+  }
+  if (ch4<1) {// && (fNegStrat==HLTv2::ReqPos || fNegStrat==HLTv2::FromGreg)) {
+    ch4=0;
+  }
+  if (ch5<1) {// && (fNegStrat==HLTv2::ReqPos || fNegStrat==HLTv2::FromGreg)) {
+    ch5=0;
+  }
+  
+  // Print out 
+  HLTOutput.clear();
+  HLTOutput.push_back(ch4*gainCorr);// amplitude 
+  HLTOutput.push_back(tsShift4); // time shift of in-time pulse
+  HLTOutput.push_back(ch5); // whatever
+
+}
+
+void HLTv2::applyXM(const std::vector<double> & inputCharge, const std::vector<double> & inputPedestal, std::vector<double> & HLTOutput) const {
+
+  std::vector<double> corrCharge;
+  fPedestalSubFxn_.Calculate(inputCharge, inputPedestal, corrCharge);
+
+  double TS35[3];
+  double TS46[3];
+  double TS57[3];
+  PulseFraction(inputCharge[3], TS35);
+  PulseFraction(inputCharge[4], TS46);
+  PulseFraction(inputCharge[5], TS57);
+
+  double a3[3] = {TS35[0], TS35[1], TS35[2]};
+  double b3[3] = {0., TS46[0], TS46[1]};
+  double c3[3] = {0., 0., TS57[0]};
+  double d3[3] = {corrCharge[3], corrCharge[4], corrCharge[5]};
+
+  double deno3 = Det3(a3, b3, c3);
+
+  double A3 = Det3(d3, b3, c3) / deno3;
+  double A4 = Det3(a3, d3, c3) / deno3;
+  double A5 = Det3(a3, b3, d3) / deno3;
+  HLTOutput.clear();
+  HLTOutput.push_back(A3);
+  HLTOutput.push_back(A4);
+  HLTOutput.push_back(A5);
+
+}
+
+constexpr Float_t HLTv2::landauFrac[];
+// Landau function integrated in 1 ns intervals
+//Landau pulse shape from https://indico.cern.ch/event/345283/contribution/3/material/slides/0.pdf
+//Landau turn on by default at left edge of time slice 
+// normalized to 1 on [0,10000]
+void HLTv2::getLandauFrac(Float_t tStart, Float_t tEnd, Float_t &sum) const{
+
+  if (abs(tStart-tEnd-25)<0.1) {
+    cout << abs(tStart-tEnd) << "???" << endl;
```

Comment:
remove cout.
If a warning message is needed, use LogWarning, LogError or a similar appropriate message